src/vscode-dts/vscode.proposed.toolProgress.d.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	/**
+	 * A progress update during an {@link LanguageModelTool.invoke} call.
+	 */
+	export interface ToolProgressStep {
+		/**
+		 * A progress message that represents a chunk of work
+		 */
+		message?: string;
+		/**
+		 * An increment for discrete progress. Increments will be summed up until 100 (100%) is reached
+		 */
+		increment?: number;
+	}
+
+	export interface LanguageModelToolInvocation<T> extends LanguageModelToolInvocationOptions<T> {
+		/**
+		 * Progress interface to report updates as the tool is running.
+		 */
+		progress: Progress<ToolProgressStep>;
```

Comment:
In my last commit I added this options-extension version as a way to implement this. Imo this is much nicer and gives us a way to keep adding editor-provided data (where `LanguageModelToolInvocationOptions` is also extension-created in `vscode.lm.invokeTool`). We already do this in `chatParticipantPrivate.d.ts` by just making `chatRequestId` and friends optional. But if we want to go back to the 3rd argument, we can revert the last commit.

I also don't know what a good name for this is so I just went with `LanguageModelToolInvocation` mirroring what we have in core. 😛

Reply:
I liked the argument variant better